src/main/java/stirling/software/SPDF/config/security/database/DatabaseService.java

```diff
@@ -0,0 +1,290 @@
+package stirling.software.SPDF.config.security.database;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.sql.DataSource;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.PathResource;
+import org.springframework.core.io.support.EncodedResource;
+import org.springframework.jdbc.datasource.init.CannotReadScriptException;
+import org.springframework.jdbc.datasource.init.ScriptException;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.stereotype.Service;
+
+import lombok.extern.slf4j.Slf4j;
+import stirling.software.SPDF.config.interfaces.DatabaseInterface;
+import stirling.software.SPDF.model.ApplicationProperties;
+import stirling.software.SPDF.model.exception.BackupNotFoundException;
+import stirling.software.SPDF.utils.FileInfo;
+
+@Slf4j
+@Service
+public class DatabaseService implements DatabaseInterface {
```

Comment:
I think this class only applies to H2 right? we need checks to skip everything here if its not H2

Reply:
And we need to test 
http://localhost:8080/database
For H2 still works as expected, and expected behaviour is probably to disable  backups for non H2 
Expectation is if its external database that database should handle any backups etc

Imports should be supported in both formats
A user should be able to
h2 -> export
and then import into postgres with that data

Reply:
Will work on this in a new ticket